DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/PostLearningAssessmentViewModelTests.cs

```diff
@@ -0,0 +1,188 @@
+        }
+
+        [Test]
+        public void Post_learning_assessment_can_be_not_locked()
+        {
+            // Given
+            const bool postLearningLocked = false;
+            var postLearningAssessment = PostLearningAssessmentHelper.CreateDefaultPostLearningAssessment(
+                plLocked: postLearningLocked
+            );
+
+            // When
+            var postLearningAssessmentViewModel =
+                new PostLearningAssessmentViewModel(postLearningAssessment, CustomisationId, SectionId);
+
+            // Then
+            postLearningAssessmentViewModel.PostLearningLocked.Should().BeFalse();
+        }
+
+        [Test]
+        public void Post_learning_assessment_assessment_status_with_no_attempts_should_be_not_attempted()
+        {
+            // Given
+            const int postLearningAttempts = 0;
+            var postLearningAssessment = PostLearningAssessmentHelper.CreateDefaultPostLearningAssessment(
+                attemptsPl: postLearningAttempts
+            );
+
+            // When
+            var postLearningAssessmentViewModel =
+                new PostLearningAssessmentViewModel(postLearningAssessment, CustomisationId, SectionId);
+
+            // Then
+            postLearningAssessmentViewModel.AssessmentStatus.Should().Be("Not attempted");
+        }
+
+        [Test]
+        public void Post_learning_assessment_attempts_information_with_attempts_can_be_passed()
+        {
+            // Given
+            const int postLearningAttempts = 4;
+            const int postLearningPasses = 2;
+            var postLearningAssessment = PostLearningAssessmentHelper.CreateDefaultPostLearningAssessment(
+                attemptsPl: postLearningAttempts,
+                plPasses: postLearningPasses,
+                maxScorePl: 90
+            );
+
+            // When
+            var postLearningAssessmentViewModel =
+                new PostLearningAssessmentViewModel(postLearningAssessment, CustomisationId, SectionId);
+
+            // Then
+            postLearningAssessmentViewModel.AssessmentStatus.Should().Be("Passed (90% - 4 attempts)");
+        }
+
+        [Test]
+        public void Post_learning_assessment_attempts_information_with_attempts_can_be_failed()
+        {
+            // Given
+            const int postLearningAttempts = 5;
+            const int postLearningPasses = 0;
+            var postLearningAssessment = PostLearningAssessmentHelper.CreateDefaultPostLearningAssessment(
+                attemptsPl: postLearningAttempts,
+                plPasses: postLearningPasses,
+                maxScorePl: 10
+            );
+
+            // When
+            var postLearningAssessmentViewModel =
+                new PostLearningAssessmentViewModel(postLearningAssessment, CustomisationId, SectionId);
+
+            // Then
+            postLearningAssessmentViewModel.AssessmentStatus.Should().Be("Failed (10% - 5 attempts)");
```

Comment:
What about if the percentage is not a whole number, rouding is probably required? We should have a test for that (and code to do the rounding if we don't already) 🙂

Reply:
It looks like `AssessAttempts.Score` where we get the score from is stored as an int so no rounding should be necessary 🙂

Reply:
Ah I see. I think I was getting confused by the `maxScorePl` argument name. Could we rename this to more accurately reflect what it represents (their highest score so far I believe)? 🙂

Reply:
Yes I agree the name is a bit misleading, I used it because it was the name of the column returned in the stored procedure. What do you think of `BestScore`?

Reply:
I like it 🙂